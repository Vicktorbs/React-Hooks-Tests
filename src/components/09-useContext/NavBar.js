import React from 'react';
import { NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark  bg-dark">
            <div className="container-fluid">
                <NavLink className="navbar-brand" to="/">useContext</NavLink>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        {/* NavLink permite identificar la ruta en la que se esta ubicado */}
                        <NavLink activeClassName="active" exact to="/" className="nav-link active" aria-current="page">Home</NavLink>
                        <NavLink activeClassName="active" exact to="/about" className="nav-link">About</NavLink>
                        <NavLink activeClassName="active" exact to="/login" className="nav-link">Login</NavLink>
                        {/* <Link to="/" className="nav-link active" aria-current="page">Home</Link>
                        <Link to="/about" className="nav-link">About</Link>
                        <Link to="/login" className="nav-link">Login</Link> */}
                    </div>
                </div>
            </div>
        </nav>
    )
}
