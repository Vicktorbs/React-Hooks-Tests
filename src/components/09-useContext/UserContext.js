import { createContext } from "react";

// Higher order component, puede ser accesado desde cualquiera de sus componentes hijos evitando la herarquia
export const UserContext = createContext(null);