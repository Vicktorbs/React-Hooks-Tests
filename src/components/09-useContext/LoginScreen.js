import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />

            <button className="btn btn-primary" onClick={ () => setUser({name: 'victor', id: 54321}) }>
                Login
            </button>
        </div>
    )
}
