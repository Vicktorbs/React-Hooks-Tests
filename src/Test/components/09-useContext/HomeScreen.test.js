import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test on conponente <HomeScreen />', () => {
    
    // Como definir un contexto en un entorno aislado
    const user = {
        name: 'Victor',
        email: 'viktorb.132@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
