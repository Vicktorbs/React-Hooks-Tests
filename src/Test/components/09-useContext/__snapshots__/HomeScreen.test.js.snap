// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on conponente <HomeScreen /> should show itself correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Victor",
   "email": "viktorb.132@gmail.com"
}
  </pre>
</div>
`;
